services/libs/aster-std/src/syscall/mmap.rs

```diff
@@ -78,17 +78,29 @@ fn mmap_anonymous_vmo(
         panic!("Unsupported mmap flags {:?} now", option);
     }
 
-    let vmo_options: VmoOptions<Rights> = VmoOptions::new(len);
-    let vmo = vmo_options.alloc()?;
-    let current = current!();
-    let root_vmar = current.root_vmar();
+    let vmo = {
+        let vmo_options: VmoOptions<Rights> = VmoOptions::new(len);
+        vmo_options.alloc()?
+    };
+
+    let mut vmar_map_options = {
+        let current = current!();
+        let root_vmar = current.root_vmar();
+        root_vmar.new_map(vmo, perms)?
+    };
 
-    let mut vmar_map_options = root_vmar.new_map(vmo, perms)?;
-    if option.flags().contains(MMapFlags::MAP_FIXED) {
+    let flags = option.flags();
+    if flags.contains(MMapFlags::MAP_FIXED) {
         vmar_map_options = vmar_map_options.offset(addr).can_overwrite(true);
+    } else if flags.contains(MMapFlags::MAP_32BIT) {
+        // TODO: support MAP_32BIT. MAP_32BIT requires the map range to be below 2GB
+        warn!("MAP_32BIT is not supported");
     }
+
     let map_addr = vmar_map_options.build()?;
+
     debug!("map addr = 0x{:x}", map_addr);
```

Comment:
There are almost the same logging logic, but with slight differences in `mmap_filebacked_vmo`.

```
    let map_addr = vm_map_options.build()?;
    trace!("map range = 0x{:x} - 0x{:x}", map_addr, map_addr + len);
```

It is unclear why the logging output should be different in the two similar methods.